e2e/common_test.go

```diff
@@ -49,3 +53,76 @@ func RunCmd(cmd *exec.Cmd) (stdoutBytes []byte, stderrBytes []byte) {
 
 	return stdout.Bytes(), stderr.Bytes()
 }
+
+type forwardedPort struct {
+	local    int
+	remote   int
+	protocol string
+}
+
+func startForwardingTenantAPI(name, namespace string, localPort int) (net.Listener, error) {
+	address, err := net.ResolveIPAddr("", "127.0.0.1")
+	port := forwardedPort{
+		local:    localPort,
+		remote:   6443,
+		protocol: "tcp",
+	}
+	if err != nil {
+		return nil, err
+	}
+	listener, err := net.ListenTCP(
```

Comment:
listener.Close() is never called. Maybe we need a `stopForwardingTenantAPI` function that the test driver calls during cleanup?